src/sentry/static/sentry/app/views/eventsV2/eventDetails/content.tsx

```diff
@@ -0,0 +1,339 @@
+import React from 'react';
+import {Params} from 'react-router/lib/Router';
+import DocumentTitle from 'react-document-title';
+import {Location} from 'history';
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import {trackAnalyticsEvent} from 'app/utils/analytics';
+import {Client} from 'app/api';
+import withApi from 'app/utils/withApi';
+import {getMessage, getTitle} from 'app/utils/events';
+import {Organization, Event} from 'app/types';
+import SentryTypes from 'app/sentryTypes';
+import getDynamicText from 'app/utils/getDynamicText';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import DateTime from 'app/components/dateTime';
+import ExternalLink from 'app/components/links/externalLink';
+import FileSize from 'app/components/fileSize';
+import {PageHeader} from 'app/styles/organization';
+import NotFound from 'app/components/errors/notFound';
+import AsyncComponent from 'app/components/asyncComponent';
+
+import EventView from '../eventView';
+import {hasAggregateField, EventQuery} from '../utils';
+import Pagination from './pagination';
+import LineGraph from './lineGraph';
+import RelatedEvents from '../relatedEvents';
+import TagsTable from '../tagsTable';
+import EventInterfaces from '../eventInterfaces';
+import LinkedIssuePreview from '../linkedIssuePreview';
+import DiscoverBreadcrumb from '../breadcrumb';
+
+const slugValidator = function(
+  props: {[key: string]: any},
+  propName: string,
+  componentName: string
+) {
+  const value = props[propName];
+  // Accept slugs that look like:
+  // * project-slug:deadbeef
+  if (value && typeof value === 'string' && !/^(?:[^:]+):(?:[a-f0-9-]+)$/.test(value)) {
+    return new Error(`Invalid value for ${propName} provided to ${componentName}.`);
+  }
+  return null;
+};
+
+type Props = {
+  organization: Organization;
+  location: Location;
+  params: Params;
+  api: Client;
+  eventSlug: string;
+};
+
+type State = {
+  event: Event | undefined;
+};
+
+class EventDetailsContent extends AsyncComponent<Props, State & AsyncComponent['state']> {
+  static propTypes: any = {
+    organization: SentryTypes.Organization.isRequired,
+    eventSlug: slugValidator,
```

Comment:
Nah, it doesn't transform the prop values. IIRC, custom proptypes return an `Error` object if the validation fails.

I think it's very uncommon to have custom proptypes nowadays 😄 . 

Since this file is only consumed by other TS files, I chose to keep these proptypes instead of removing them altogether since I'd like to keep the `slugValidator` around for a bit until Discover2 settles a bit more.